File: _demo/sfml/system/time/cpp/time.cpp
#include <SFML/System/Time.hpp>
#include <stdio.h>
int main(void) {
    sf::Time time = sf::Time();
    printf("time.asSeconds:%f\n", time.asSeconds());
    printf("time.asMilliseconds:%d\n", time.asMilliseconds());
    printf("time.asMicroseconds:%ld\n", time.asMicroseconds());
    printf("sf::Time::Zero.asSeconds:%f\n", sf::Time::Zero.asSeconds());
    printf("sf::Time::Zero.asMilliseconds:%d\n", sf::Time::Zero.asMilliseconds());
    printf("sf::Time::Zero.asMicroseconds:%ld\n", sf::Time::Zero.asMicroseconds());
    return 0;
}

File: c/sfml/system/time/time.go
package time

import (
	_ "unsafe"
)

const (
	LLGoPackage = "link: $(pkg-config --libs sfml-system); -lsfml-system"
)

type Time struct {
	m_microseconds int64
}

//go:linkname Init C._ZN2sf4TimeC1Ev
func Init(time *Time) {
	time.m_microseconds = 0
}

//go:linkname AsSeconds C._ZNK2sf4Time9asSecondsEv
func AsSeconds(time *Time) float32

//go:linkname AsMilliseconds C._ZNK2sf4Time14asMillisecondsEv
func AsMilliseconds(time *Time) int32

//go:linkname AsMicroseconds C._ZNK2sf4Time14asMicrosecondsEv
func AsMicroseconds(time *Time) int64
